harfbuzz/src/unicode_funcs.rs

```diff
@@ -0,0 +1,352 @@
+// Copyright 2023 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use crate::sys::*;
+use crate::Error;
+use std::ffi::c_void;
+
+/// An object to map from code points to general category properties.
+pub trait GeneralCategoryFunc {
+    /// Given a code point, return the general category as a
+    /// [`hb_unicode_general_category_t`].
+    fn general_category(&self, ch: u32) -> hb_unicode_general_category_t;
```

Comment:
Thought: Perhaps these return values should not use `harfbuzz-sys` types such that these traits can be implemented without a `harfbuzz-sys` dependency (see https://github.com/unicode-org/icu4x/issues/3175)